src/dataflow/src/render/context.rs

```diff
@@ -490,6 +490,13 @@ where
     ) {
         mfp.optimize();
         let mfp_plan = mfp.into_plan().unwrap();
+        // This path results in a slightly smaller and faster
+        // dataflow graph, and is intended to fix
+        // https://github.com/MaterializeInc/materialize/issues/10507
+        if mfp_plan.is_identity() && !matches!(key_val, Some((_, Some(_)))) {
```

Comment:
I find the `!matches` to use too many negations, what about this:
```
if mfp_plan.is_identity() {
   if let Some((key, None)) = key_val {
       return self.as_specific_collection(key.as_deref())
   }
}
```

Reply:
That is logically different I think. `None` matches the macro and not this, and .. idk if is or isn't appropriate for what follows.